### PR TITLE
AlignmentMatrixControl: simplify styles and markup

### PR DESCRIPTION
**Pull request by ciampo**
## What?

In preparation for #64723

Simplify `AlignmentMatrixControl` styles, rewrite `AlignmentMatrixControl.Icon` using SVG, add better focus styles and support for Windows High Contrast mode.

## Why?

While working on #64723, I realized that `AlignmentMatrixControl` and `AlignmentMatrixControl.Icon` have some unnecessarily complex code, which made the refactor in #64723 harder than it should have been.

## How?

- `AlignmentMatrixControl`:
  - Instead of using JS logic, the active `Composite` item is styled based on the `data-active-item` attribute;
  - Added a focus ring only when using the keyboard;
  - The component is now fully compatible with Windows High Contrast mode (see attachments below);
  - Stopped using deprecated `COLORS` variables;
- `AlignmentMatrixControl.Icon`:
  - Rewrote completely using SVG — it is an icon after all, and I didn't see valid reasons for using non-svg markup. Using SVG also means that all additional custom styles are unnecessary.
  - The component is now fully compatible with Windows High Contrast mode (see attachments below)
  - The `size` prop on `AlignmentMatrixControl.Icon` is deprecated, in favor of using `width` and `height`. This is mostly there to reflect that this icon is supposed to be used in combination with the `<Icon />` component (as shown in the Storybook examples and done in the editor), which accepts a `size` prop itself, and internally sets the `width` and `height` props on the SVG.
  - Previously, the scaling of `AlignmentMatrixControl.Icon` was done by applying`transform: scale()` via CSS — which meant that the footprint of the icon in the layout flow would stay the same (`24px`) regardless of the `size`. With the changes in this PR, the `AlignmentMatrixControl.Icon` correctly occupies an amount of space equal to its `width` and `height`.
  - The `AlignmentMatrixControl.Icon` now accepts SVG-related HTML attributes, while previously it accepted HTML `div` ones (this is technically a breaking change, although I don't see how consumers would be affected)

## Testing Instructions

- Make sure unit tests pass
- Check the components in Storybook and make sure they look and behave like on `trunk`
  - Check that the version from this PR shows a focus ring around the selected `AlignmentMatrixControl` item when using the keyboard (and not when using mouse);
  - Check that both `AlignmentMatrixControl` and `AlignmentMatrixControl.Icon` can be perceived and interacted with in Windows High Contrast mode (this can also be testing by [emulating the `forced-colors` media query](https://devtoolstips.org/tips/en/emulate-forced-colors/))
- In the editor, add a Cover block and add an image:
	- In the block toolbar, notice the `AlignmentMatrixControl.Icon` being used in a toolbar item. Clicking it will open a popover containing a `AlignmentMatrixControl` component;
	- Interact with both components, make sure they work as expected
	- Follow the same instructions shared above about testing focus rings and Windows High Contrast mode

## Screenshots or screencast 

| Before (trunk) | After (this PR) |
|---|---|
| <video src="https://github.com/user-attachments/assets/b0c98fbe-4ed0-4ebd-9db2-d4c612f3c102" /> | <video src="https://github.com/user-attachments/assets/5a2d6356-f96d-4273-b7b7-08be5fcad4e8" /> |
| Screenshot 2024-08-27 at 15 23 06 | Screenshot 2024-08-27 at 15 24 29 |
| Screenshot 2024-08-27 at 15 24 03 | Screenshot 2024-08-27 at 15 25 12 |
| <video src="https://github.com/user-attachments/assets/b365e431-1e75-4c9a-bdd7-b43340088098" /> | <video src="https://github.com/user-attachments/assets/33923fc4-e5ed-40ef-a6d7-ea56b2247dd5" /> |

### CODE REVIEW

File: packages/components/src/alignment-matrix-control/icon.tsx [hunk 1]
```diff
@@ -3,59 +3,77 @@
...
-		className
-	);
-
-	const styles = {
-		...style,
-		transform: `scale(${ scale })`,
-	};
+}: WordPressComponentProps< AlignmentMatrixControlIconProps, 'svg', false > ) {
+	const computedWidth = size ?? width ?? BASE_SIZE;
+	const computedHeight = size ?? width ?? BASE_SIZE;
```
Comment (mirka) [inline packages/components/src/alignment-matrix-control/icon.tsx]: ?

```suggestion
	const computedHeight = size ?? height ?? BASE_SIZE;
```

File: packages/components/src/alignment-matrix-control/types.ts [hunk 2]
```diff
@@ -44,11 +44,32 @@ export type AlignmentMatrixControlIconProps = Pick<
...
+	 */
+	height?: number;
+	/**
+	 * _Note: this prop is deprecated. Use the `width` and `height` props, or the
+	 * `size` prop on the parent `Icon` component instead_
+	 *
+	 * The size of the icon.
+	 * @ignore
+	 * @default 24
+	 */
```
Comment (mirka) [inline packages/components/src/alignment-matrix-control/types.ts]: The `size` prop was simply more ergonomic than `width`/`height`, so it feels weird that we're highlighting the `width`/`height` props now. What do you think about _not_ adding explicit types and descriptions here? They will be added in "silently" through the `WordPressComponentProps<>`.

I am ok with soft deprecating the `size` prop — "just use the `size` prop on `Icon`" makes sense.

**Review by mirka — APPROVED** [review]
Really nice cleanup, and great bug fix with the Windows High Contrast mode!

File: packages/components/src/alignment-matrix-control/icon.tsx (hunk 1 above)
Comment (ciampo) [reply packages/components/src/alignment-matrix-control/icon.tsx]: Nothing to see here 👀  🤣

File: packages/components/src/alignment-matrix-control/types.ts (hunk 2 above)
Comment (ciampo) [reply packages/components/src/alignment-matrix-control/types.ts]: Makes sense, even more so for the fact that `width` and `height` are legit SVG props. I applied your suggestions

**Comment by t-hamano** [comment]
Sorry for the late reply. Just wanted to let you know that it works as expected in Windows high contrast mode too 👍

https://github.com/user-attachments/assets/ac986c48-afd6-4703-92d4-69a4638f427f

https://github.com/user-attachments/assets/ae04eaec-a46c-42ae-815d-29c34ce1fbb8

**Pull request merged**